### PR TITLE
Add context manager to collect Requests/Responses.

### PR DESCRIPTION
**Pull request by danielballan**
See addition to docs for example. This seems useful for digging into issues where there are too many log messages.

Is this a good name, @dylanmcreynolds?

### CODE REVIEW

**Comment by danielballan** [comment]
Example usage in the wild:

```py
In [12]: with telemetry() as t:
    ...:     run['primary']['data'].read(['zpssx', 'zpssy', 'zpssz'])
    ...: 

In [13]: t.requests
Out[13]: 
[<Request('GET', 'http://local-tiled-app/api/node/search/f07e3065-ab92-4b20-a702-ef61ed164dbc?filter%5Blookup%5D%5Bcondition%5D%5Bkey%5D=primary&sort=_')>,
 <Request('GET', 'http://local-tiled-app/api/node/search/f07e3065-ab92-4b20-a702-ef61ed164dbc/primary?filter%5Blookup%5D%5Bcondition%5D%5Bkey%5D=data&sort=_')>,
 <Request('GET', 'http://local-tiled-app/api/array/block/f07e3065-ab92-4b20-a702-ef61ed164dbc/primary/data/coords/time/variable?block=0&expected_shape=5625')>,
 <Request('GET', 'http://local-tiled-app/api/node/full/f07e3065-ab92-4b20-a702-ef61ed164dbc/primary/data?format=application%2Fvnd.apache.arrow.file&field=zpssx&field=zpssy&field=zpssz')>]

In [14]: t.responses
Out[14]: 
[<Response [200 OK]>,
 <Response [200 OK]>,
 <Response [200 OK]>,
 <Response [200 OK]>]
```

**Comment by danielballan** [comment]
The word "telemetry" came into my head because I was focused on looking at `server-timing` headers in the responses, but now I think a much better name for this context manager would be `record`.

File: tiled/_tests/test_telemetry.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+tree = MapAdapter({})
+
+
+def test_telemetry():
+    "Very basic exercise of telemetry"
+    client = from_tree(tree)
+
+    with telemetry() as t:
+        repr(client)  # trigger a request
+    assert t.requests
```
Comment (dylanmcreynolds) [inline tiled/_tests/test_telemetry.py:13]: I think you're missing an indent.

File: docs/source/how-to/client-logger.md [hunk 2]
```diff
@@ -97,3 +97,15 @@ from tiled.client import hide_logs
...
+
+## Collect requests and responses for advanced profiling
+
+```python
+from tiled.client import telemetry
+
+with telemetry() as t:
+    ...
+
+t.requests  # list of Requests
```
Comment (dylanmcreynolds) [inline docs/source/how-to/client-logger.md:109]: I looks like you're missing an indent here?

**Review by dylanmcreynolds — COMMENTED** [review]
Wow, cool!

**Comment by danielballan** [comment]
Renamed in brainstorming session with @dylanmcreynolds from `telemetry()` to `record_history()`. The term "telemetry" isn't wrong but "record_history" is plainer.

**Pull request merged**